front/src/modules/ui/input/hooks/useLazyLoadIcon.ts

```diff
@@ -2,20 +2,21 @@ import { useEffect, useState } from 'react';
 
 import { IconComponent } from '@/ui/display/icon/types/IconComponent';
 
+import { useLazyLoadIcons } from './useLazyLoadIcons';
+
 export const useLazyLoadIcon = (iconKey: string) => {
+  const { isLoadingIcons, icons } = useLazyLoadIcons();
   const [Icon, setIcon] = useState<IconComponent | undefined>();
   const [isLoadingIcon, setIsLoadingIcon] = useState(true);
 
   useEffect(() => {
     if (!iconKey) return;
 
-    import(`@tabler/icons-react/dist/esm/icons/${iconKey}.js`).then(
-      (lazyLoadedIcon) => {
-        setIcon(lazyLoadedIcon.default);
-        setIsLoadingIcon(false);
-      },
-    );
-  }, [iconKey]);
+    if (!isLoadingIcons) {
```

Comment:
@thaisguigon FYI I've changed this as this is causing performance issues. In the end, I think lazy loading all icons is enough, we will pretty much use 20% of them on the long run anyway. What's important is that they don't get in the way of the first print